packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2149,7 +2149,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     final TextPosition position = _textPainter.getPositionForOffset(globalToLocal(_lastTapDownPosition! - _paintOffset));
     final TextRange word = _textPainter.getWordBoundary(position);
     late TextSelection newSelection;
-    if (position.offset - word.start <= 1) {
```

Comment:
Not entirely sure why the original logic check for <= 1. I tested native iOS and it looks like you need to click at the very beginning of the word to move the caret to start, so I think the  change matches the native behavior.